src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -460,6 +465,34 @@ private static Details getDetailsForSwitch(DetailAST switchNode) {
             return new Details(lcurly, rcurly, nextToken, true);
         }
 
+        /**
+         * Collects details about case statements.
+         *
+         * @param caseNode case statement to gather details about
+         * @return new Details about given case statement
+         */
+        private static Details getDetailsForCase(DetailAST caseNode) {
+            final DetailAST caseParent = caseNode.getParent();
+            final int parentType = caseParent.getType();
+            final DetailAST ast = caseParent.findFirstToken(TokenTypes.SLIST);
+
+            // case group may have SLIST but not lcurly
+            final Optional<DetailAST> caseGroupLcurly = Optional.ofNullable(ast)
+                    .filter(node -> node.getFirstChild().getType() == TokenTypes.SLIST)
+                    .map(DetailAST::getFirstChild);
+
+            final Optional<DetailAST> switchRuleLcurly = Optional.ofNullable(ast)
+                    .filter(node -> parentType == TokenTypes.SWITCH_RULE);
+
+            final DetailAST lcurly = caseGroupLcurly.orElse(switchRuleLcurly.orElse(null));
+
+            DetailAST rcurly = null;
+            if (lcurly != null && !isSwitchExpression(caseParent)) {
+                rcurly = lcurly.getLastChild();
+            }
+            return new Details(lcurly, rcurly, getNextToken(caseParent), true);
```

Comment:
There is a few ways we can do this, but we can clean this up a bit more and do the same for rcurly:
```suggestion
          final DetailAST statementList = caseParent.findFirstToken(TokenTypes.SLIST);

            final Optional<DetailAST> lcurly;
            if (parentType == TokenTypes.SWITCH_RULE) {
                lcurly = Optional.ofNullable(statementList);
            }
            else {
                lcurly = Optional.ofNullable(statementList)
                        .map(DetailAST::getFirstChild)
                        .filter(node -> node.getType() == TokenTypes.SLIST);
                        
            }

            final DetailAST rcurly = lcurly.map(DetailAST::getLastChild)
                    .filter(child -> !isSwitchExpression(caseParent))
                    .orElse(null);
            
            return new Details(lcurly.orElse(null), rcurly, getNextToken(caseParent), true);
```

Reply:
@nrmancuso very good done